internal/resources/cluster/resource_cluster.go

```diff
@@ -295,68 +300,99 @@ func resourceClusterDelete(_ context.Context, d *schema.ResourceData, m interfac
 	return diags
 }
 
-func resourceClusterInPlaceUpdate(ctx context.Context, d *schema.ResourceData, m interface{}) (diags diag.Diagnostics) {
-	config := m.(authctx.TanzuContext)
-
-	updateRequired := false
+func withTKGmVsphereVersionUpdate(d *schema.ResourceData, cluster *clustermodel.VmwareTanzuManageV1alpha1ClusterCluster) {
+	if d.HasChange(helper.GetFirstElementOf(SpecKey, tkgVsphereClusterKey, distributionKey, versionKey)) {
+		newVersion := d.Get(helper.GetFirstElementOf(SpecKey, tkgVsphereClusterKey, distributionKey, versionKey))
+		cluster.Spec.TkgVsphere.Distribution.Version = newVersion.(string)
+		updateAvailable = true
 
-	switch {
-	case common.HasMetaChanged(d):
-		fallthrough
-	case d.HasChange(helper.GetFirstElementOf(SpecKey, clusterGroupKey)):
-		updateRequired = true
+		log.Printf("[INFO] updating TKGm vSphere workload cluster version to %s", newVersion.(string))
 	}
+}
 
-	if !updateRequired {
-		return diags
+func withTKGsVsphereVersionUpdate(d *schema.ResourceData, cluster *clustermodel.VmwareTanzuManageV1alpha1ClusterCluster) {
+	if d.HasChange(helper.GetFirstElementOf(SpecKey, tkgServiceVsphereKey, distributionKey, versionKey)) {
+		newVersion := d.Get(helper.GetFirstElementOf(SpecKey, tkgServiceVsphereKey, distributionKey, versionKey))
+		cluster.Spec.TkgServiceVsphere.Distribution.Version = newVersion.(string)
+		updateAvailable = true
+
+		log.Printf("[INFO] updating TKGs workload cluster version to %s", newVersion.(string))
 	}
+}
 
-	getResp, err := config.TMCConnection.ClusterResourceService.ManageV1alpha1ClusterResourceServiceGet(constructFullname(d))
-	if err != nil {
-		return diag.FromErr(errors.Wrapf(err, "Unable to get Tanzu Mission Control cluster entry, name : %s", d.Get(clusterNameKey)))
+func withClusterGroupUpdate(d *schema.ResourceData, cluster *clustermodel.VmwareTanzuManageV1alpha1ClusterCluster) {
+	if d.HasChange(helper.GetFirstElementOf(SpecKey, clusterGroupKey)) {
+		newClusterGroupName := d.Get(helper.GetFirstElementOf(SpecKey, clusterGroupKey))
+		if newClusterGroupName.(string) != "" {
+			cluster.Spec.ClusterGroupName = newClusterGroupName.(string)
+		}
+
+		updateAvailable = true
```

Comment:
updateAvailable and the log statement should be inside the if condition... Even better flip the if condition and return

Reply:
Done